data_collection/web/guides.rst

```diff
@@ -40,8 +40,51 @@ Example::
 
 .. todo:: Explain how to change page URL and title for virtual page views or link to SPA instructions page.
 
+User ID
+-------
+
+`User ID` is an additional parameter, that allows you to aggregate data. When set you will be 
+able to search through sessions by this parameter, fiter reports through it or create Multi attribution reports
+using `User ID`. You can :ref:`learn more about User ID here<https://help.piwik.pro/support/getting-started/userid/>` 
+To set up `USer ID` with your JavaScript Tracking snippet follow this guide.
+
+To set up `User ID` parameter add a call to :ref:`setUserId<jtc-api-setUserId>` to your tracking code:
+
+.. code-block:: javascript
+
+    _paq.push(["setUserId", "user-name@example.com"]);
+
+.. note ::
+
+    #. Invoking ``setUserId`` won't send any tracking request. To add `User ID` to tracked data, you have to call ``setUserId`` before :ref:`trackPageView<jtc-api-trackPageView>`. It does not have to be the first one. Another way is to send ping request with :ref:`ping<jtc-api-ping>`, after setting the `User ID`.
+    #. `User ID` can't be longer than 1024 bytes, it will be 1024 characters unless you use only ASCII characters. When using Unicode check the length of user identifier, so you won't exceed 1024 bytes.
+    #. `User ID` should be a uniquie value for each user. Otherwise metrics for different users might be merged in the reports.
+    #. Usually a `User ID` value is an user email, because this is the identifier that users use to log in to a website and it fulfils above requirements.
+
+It is a good practice to remove value of `User ID`, when the user logs out. Otherwise `User ID` value might affect session of other users, if they share the same device. To remove `User ID` value call :ref:`resetUserId<jtc-api-resetUserId>`.
+
+.. code-block:: javascript
+
+    _paq.push(["resetUserId"]);
+
+Full abstract example, might look like this:
+
+.. code-block:: javascript
 
+    var user = getUserData();
+    if (user.isLogged) {
+      _paq.push(["setUserId", user.login]);
+    } else {
+      _paq.push(["resetUserId"]);
+    }
+
+.. warning::
+
+    Do not unset `User ID` byt setting it to some seemingly empty value, like ``_paq.push(["setUserId", " "]);`` or ``_paq.push(["setUserId", ""]);``. This way some value might be still send to Collecting & Processing Pipeline. What seems to be an empty value to a human, might be not true for a machine. Only using ``resetUserId`` will guarantee that no `User ID` value will be send.
```

Comment:
Internally it is set to empty string ([ref](https://github.com/PiwikPRO/PPMS-Trucker-JS/blob/master/src/piwik.js#L4768)) and we [check](https://github.com/PiwikPRO/PPMS-Trucker-JS/blob/master/src/piwik.js#L3208) for empty value in code before using it.
```
this.resetUserId = function() {
    configUserId = '';
};
```

it's internal implementation but I expect that we'll always check for string values.
If we want to leave this warning, then maybe this will work better:

```suggestion
    Do not reset `User ID` by setting it to arbitrary empty value, like ``_paq.push(["setUserId", " "]);`` or ``_paq.push(["setUserId", ""]);``. While some of those examples may work in current version, only using ``resetUserId`` will guarantee that `User ID` value will be correctly removed in the future.
```

Reply:
Not like this, we want users to use `resetUserId`, not to try to set empty value for themselves. We already had cases that user set some space or invisible unicode and told us that `User ID` do not work. Even if in reality `resetUserId` sets it t empty string, we want customers to use this very function to turn off `uid` parameter.